### PR TITLE
feat(core): add a11y/positive-tabindex rule

### LINKED ISSUE #582: a11y: flag positive tabindex values — tabindex above 0 hijacks the tab order for the whole page
## Problem

A `tabindex` greater than 0 puts the element ahead of every naturally-ordered element on the page, so a single `tabindex="1"` reorders keyboard navigation globally and gets worse with every element added later. The universally-agreed guidance is that only `0` (join the natural order) and `-1` (programmatically focusable) are safe. This is one of the highest-confidence a11y checks that exists — a positive value is essentially never intentional-and-correct — and it is a conspicuous hole next to the ARIA rules the a11y category already has.

## Proposed rule

`a11y/positive-tabindex` — report elements with a literal `tabindex` attribute whose value parses to an integer > 0.

- Literal values only; expression-valued `tabindex` is unknowable and passes (same convention as `isInteractiveElement` in `packages/core/src/rules/a11y/interactive.ts`, which already parses literal tabindex — reuse that parsing).
- Severity: `warning`. Recommendation: use `tabindex="0"` and put elements in DOM order, or `-1` for programmatic focus.

Reference: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex#accessibility_concerns

## Notes

- Standard new-rule checklist: register in `packages/core/src/rules/index.ts`, kitchen-sink sample, en/ja docs, regenerate rules index + skills.
- No new dependencies; the literal-tabindex parsing convention already exists in `interactive.ts`.

### PR DESCRIPTION
**Pull request by oekazuma**
## Summary

Adds `a11y/positive-tabindex` (warning): flags elements with a literal `tabindex` whose value parses to an integer above 0.

A positive tabindex puts the element ahead of every naturally-ordered element on the page, so a single `tabindex="1"` reorders keyboard navigation globally and gets worse with every element added later. Only `0` (join the natural order) and `-1` (programmatically focusable) are safe — a positive value is essentially never intentional-and-correct.

## Detection

- Literal values only; an expression-valued `tabindex` is unknowable and passes.
- Blank or non-numeric values are invalid HTML, ignored by browsers, and skipped.
- The literal-tabindex parsing is extracted from the interactive-element classification (`literalTabindexValue` in `interactive.ts`) and shared rather than duplicated.
- Built on the existing `ElementFact` walk — no new parser code.

## Included

- Rule registration, unit tests, kitchen-sink specimen with static/rendered expectations
- en/ja rule docs, regenerated rules index and skills digests, translation stamps, changeset

Closes #582

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added an accessibility rule that warns when elements use literal positive `tabindex` values.
  * Supports detection across standard and SVG elements while ignoring dynamic expressions.
  * Recommends `tabindex="0"` or `tabindex="-1"` to preserve logical keyboard navigation.

* **Documentation**
  * Added English and Japanese guidance, rule catalog entries, examples, and accessibility recommendations.

* **Tests**
  * Added coverage for valid, invalid, dynamic, and edge-case `tabindex` values.

### CODE REVIEW

**Comment by oekazuma** [comment]
@coderabbitai review

**Comment by oekazuma** [comment]
@coderabbitai review

File: docs/src/content/docs/rules/a11y/positive-tabindex.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+title: a11y/positive-tabindex · Positive tabindex
+description: A tabindex above 0 puts the element ahead of every naturally-ordered element on the page — a single tabindex="1" reorders keyboard navigation globally.
+---
+
+**Severity:** warning · **Category:** a11y
+
+## What it checks
+
+Flags an element with a literal `tabindex` attribute whose value parses to an integer greater than 0:
```
[the remark this answers is not in the document]
Comment (oekazuma) [reply docs/src/content/docs/rules/a11y/positive-tabindex.md]: Fixed in 87a4f03b: the docs (en+ja) and the changeset now say "finite number greater than 0" — the Number()-based parse does flag 1.5 — and the changeset says an expression-valued tabindex is skipped rather than passed.

File: docs/src/content/docs/rules/a11y/positive-tabindex.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+## Why it matters
+
+Elements with a positive `tabindex` come before every naturally-ordered element on the page, so a single `tabindex="1"` reorders keyboard navigation for the whole document — and the damage compounds: every element added later without a positive `tabindex` sorts after it, no matter where it sits visually. The tab order can then diverge from the visual order, which WCAG 2.4.3 (Focus Order) requires to stay meaningful. The universally-agreed guidance is that only `0` and `-1` are safe values; a positive value is essentially never intentional-and-correct.
+
+## How to fix
+
+Put elements in DOM order and let the natural tab sequence do the work, using `tabindex="0"` only to add a non-interactive element to it:
+
+```svelte
+<div tabindex="0" role="button">In the natural order</div>
```
[the remark this answers is not in the document]
Comment (oekazuma) [reply docs/src/content/docs/rules/a11y/positive-tabindex.md]: Fixed in 87a4f03b: both examples now show a scrollable region (role="region" + aria-label), the canonical legitimate tabindex="0" use that needs no activation handlers, with a note pointing anything clickable at a native <button>.

File: packages/core/src/rules/a11y/interactive.ts [hunk 3]
```diff
@@ -88,15 +88,27 @@ export function isInteractiveElement(tag: string, attrs: ElementAttr[]): boolean
...
   return hasRoleIn(attrs, INTERACTIVE_ROLES);
 }
 
+/**
+ * Numeric value of a literal `tabindex` per `Number()` semantics, or undefined when blank
+ * or unparsable. A blank tabindex is invalid HTML and ignored by browsers (`Number('')`
+ * would coerce to 0). Deliberately NOT HTML's rules-for-parsing-integers: `Number()` is
+ * what the Svelte compiler's own a11y_positive_tabindex check uses, and disagreeing with
+ * the user's build would make the rule noise (the cost is edge divergence from browsers,
+ * e.g. `"1abc"` — browser tabIndex 1 — parses to undefined here). Shared by the
```
[the remark this answers is not in the document]
Comment (oekazuma) [reply packages/core/src/rules/a11y/interactive.ts]: Fixed in 2d6ad063 — the finite-only behavior is kept deliberately: browsers ignore `tabindex="Infinity"`/overflowing values (HTML integer parsing rejects them), so warning there would be a compiler-side false positive. The helper docblock, test wording, and both Overlap sections now record it as a second deliberate divergence instead of claiming exact parity.

File: docs/src/content/docs/rules/a11y/positive-tabindex.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+Use `tabindex="-1"` for elements you focus programmatically (e.g. a skip-link target or a dialog):
+
+```svelte
+<div tabindex="-1" bind:this={panel}>Focused from code</div>
+```
+
+## Limitations
+
+Only literal `tabindex` values on native elements are covered. An expression-valued `tabindex`, a spread attribute that supplies one, and a dynamic tag via `<svelte:element>` are out of static reach and are not flagged.
```
[the remark this answers is not in the document]
Comment (oekazuma) [reply docs/src/content/docs/rules/a11y/positive-tabindex.md]: Fixed in 2d6ad063 — the Limitations now say "statically named elements" and call out that custom elements like `<my-widget tabindex="1">` are flagged.

File: docs/src/content/docs/ja/rules/a11y/positive-tabindex.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+プログラムからフォーカスする要素（スキップリンクのターゲットやダイアログなど）には `tabindex="-1"` を使います:
+
+```svelte
+<div tabindex="-1" bind:this={panel}>Focused from code</div>
+```
+
+## 制限事項
+
+対象になるのはネイティブ要素のリテラルな `tabindex` 値だけです。式による `tabindex`、それを供給しうるスプレッド属性、`<svelte:element>` による動的タグは静的解析の範囲外のため検出されません。
```
[the remark this answers is not in the document]
Comment (oekazuma) [reply docs/src/content/docs/ja/rules/a11y/positive-tabindex.md]: Fixed in 2d6ad063 — ja 側も「静的に名前が決まる要素」に修正し、カスタム要素も検出される旨を明記しました。
